primefaces/src/main/java/org/primefaces/model/JpaLazyDataModel.java

```diff
@@ -158,12 +160,18 @@ protected void applyFilters(CriteriaBuilder cb,
                     continue;
                 }
 
-                String filterValue = filter.getFilterValue().toString();
                 Field filterField = LangUtils.getFieldRecursive(entityClass, filter.getField());
-                Object convertedFilterValue = convertToType(filterValue, filterField.getType());
+                Object filterValue = filter.getFilterValue();
+                Object convertedFilterValue;
+                if (filterValue.getClass().isArray() || filterValue instanceof Collection) {
```

Comment:
In other spots in the code we use the safer `Collection.class.isAssignableFrom(filterValue.getClass())` so maybe...

```java
Class filterClass = filterValue.getClass();
if (filterClass.isArray() || ollection.class.isAssignableFrom(filterClass )) {
```

Reply:
Corrected.